spec/index.bs

```diff
@@ -730,51 +730,47 @@ For each [=subresource=] of a [=physical resource=], its set of
 Issue: This section will need to be revised to support multiple queues.
 
 On the [=Queue timeline=], there is an ordered sequence of [=usage scopes=].
-Each item on the timeline is contained within exactly one scope.
 For the duration of each scope, the set of [=internal usage=] flags of any given
 [=subresource=] is constant.
 A [=subresource=] may transition to new usages at the boundaries between [=usage scope=]s.
 
 This specification defines the following [=usage scopes=]:
 
-- Outside of a pass (in {{GPUCommandEncoder}}), each command is one usage scope
+- Outside of a pass (in {{GPUCommandEncoder}}), each (non-state-setting) command is one usage scope
     (e.g. {{GPUCommandEncoder/copyBufferToTexture()}}).
-- In a compute pass, each command is one usage scope.
+- In a compute pass, each non-state-setting command is one usage scope.
```

Comment:
This is a corner case, but maybe we need to clarify these situations: if there is no non-state-setting command, the entire compute pass is one usage scope. And if state-setting command is not the last command before pass end, then the previous state-setting command to the end of the pass is also a usage scope.  For render pass, the entire pass is always a usage scope regardless of draw call exists in the pass. So there is no such corner case in render pass.

Reply:
> if there is no non-state-setting command, the entire compute pass is one usage scope

AIUI, there are zero usage scopes in this compute pass.

> And if state-setting command is not the last command before pass end, then the previous state-setting command to the end of the pass is also a usage scope

Same, there would be no tail usage scope that these trailing state-setting commands affect.

Reply:
They are really trivial and not useful at all. They are only useful to clarify some situations for resource validation. So maybe we can say that in a note?

Reply:
OK. It makes sense too. if state-setting command is not issued, then we don't need to do resource usage validation at all. Then I may want to say that we can state that for render pass for the same sake: if state-setting command is not issues at all, then we don't need to do resource validation for the render pass.

Reply:
The render pass is a usage scope regardless of whether it has a draw. A compute pass is never a usage scope, only dispatches are usage scopes. So if there's no dispatch, there's no usage scope and no usage scope validation occurs.

Reply:
I may argue that resource usage validation is actually triggered by a command like draw/dispatch which starts a pipeline at hardware side and may use those resources. In WebGL/OpenGL, we also have feedback loop check for such things. The only difference is that we introduced the concept of "pass" and expanded resource usage validation from a draw call to a pass for render. And a pass can include multiple draw calls, you know. Per my understanding, this is because a pipeline at hardware side for the corresponding draw call can start before its previous pipeline ends in modern GPU, making resources conflicts across different pipelines at hardware side (and across draw calls at application side). So the new usage validation mechanism can fit the modern GPU design better than WebGL/OpenGL's design (which assumes the second pipeline at hardware side starts after the first pipeline ends, and no resource conflicts across different draw calls). 
I mean, it is the draw/dispatch commands that trigger resource usage conflict under the hood. So it also make sense that we don't validate resource usages if no draw/dispatch is called. But your saying is also literally correct: resource usage validation in render is triggered by endPass, and resource usage validation in compute is triggered by dispatch call. And the situation that no draw call in render pass is not useful at all. So, the current statement is acceptable for me too.

Reply:
We could have defined things so that shadowed bindings don't count toward usages. But instead we say that setBindGroup etc count toward the usage scope directly, not the current state at draw time. By those rules it doesn't matter whether there's a draw or not, and it doesn't make sense to add a specific rule that says "if there are no non-state-setting commands, skip usage scope validation". If we wanted that to be the case we would make it so that only non-state-setting commands (and the state they capture) count toward the usage scope. That's less strict than what we have now so we could still open it up someday in the future if developers have issues with these rules.

Reply:
Ack.